Sources/BlueSteel/coding/JsonAvroDecoder.swift

```diff
@@ -61,10 +61,10 @@ open class JsonAvroDecoder : AvroDecoder {
                 return AvroValue.avroUnion(schemaOptions: subSchemas, index: index, newValue)
 
             case .avroRecord(_, let fieldSchemas):
-                let fields: [(String, AvroValueConvertible)] = try fieldSchemas
+                let fields = try fieldSchemas
                     .filter { result.keys.contains($0.name) }
-                    .map { field in (field.name, try decode(any: result[field.name]!, as: field.schema)) }
-                return Dictionary(uniqueKeysWithValues: fields)
+                    .reduce(into: [:]) { $0[$1.name] = try decode(any: result[$1.name]!, as: $1.schema) }
```

Comment:
Could you please use named variables here for clarity? In that case the `fields` variable does not need to be used, the result can just directly be returned.